### PR TITLE
firefox-wrapper: remove dead npapi plugin code

### PR DESCRIPTION
**Pull request by xaverdh**
###### Motivation for this change
Support for `npapi` plugins (except for flash) was dropped from `firefox` some time ago. None of the versions in `nixpkgs` actually supports them.
So remove support for them from the wrapper. 
 
###### Things done

* Verified that the hash does not change when building with default options
* Tested that building with one of the removed `nixpkgs` options throws an error as expected

### CODE REVIEW

**Comment by xaverdh** [comment]
Most of the plugins in `pkgs/applications/networking/browsers/mozilla-plugins/` can probably be removed as well. Not sure if something else is still using them though.

**Review by lheckemann — APPROVED** [review]

File: pkgs/applications/networking/browsers/firefox/wrapper.nix [hunk 1]
```diff
@@ -38,32 +35,25 @@ let
...
+            "jre"
+            "icedtea"
+            "enableGoogleTalkPlugin"
+            "enableFriBIDPlugin"
+            "enableGnomeExtensions"
+            "enableBluejeans"
+            "enableAdobeReader"
+          ];
+        in if removed != []
+           then throw "Your configuration mentions ${lib.concatMapStringsSep ", " (p: browserName + "." + p) removed}. All plugin related options have been removed since Firefox no longer supports npapi plugins"
```
Comment (worldofpeace) [inline pkgs/applications/networking/browsers/firefox/wrapper.nix:54]: A link or a date would be nice.

Comment (xaverdh) [reply pkgs/applications/networking/browsers/firefox/wrapper.nix:54]: Will add a link

**Comment by xaverdh** [comment]
After looking through `pkgs/applications/networking/browsers/mozilla-plugins/` and some use of grep, I am fairly sure that everything there except for the `flashplayer` is unused in nixpkgs. Should I add a commit to this pr for removing them, or over a new pr for that?

**Comment by lheckemann** [comment]
IMHO add it to this PR.

**Comment by rycee** [comment]
Sorry, I missed this PR before. It would have been great if the `throw` could have been a `warn` for some period of time to avoid hard breaking of existing configurations.

**Comment by xaverdh** [comment]
> Sorry, I missed this PR before. It would have been great if the `throw` could have been a `warn` for some period of time to avoid hard breaking of existing configurations.

You think its worth changing that still?

**Comment by rycee** [comment]
I think it should be fine to keep the `throw` unless more complaints arise. I've removed the options from HM so people's configurations should now build OK again. But for future changes like this it would good to have a grace period during which a warning is printed instead of immediately going for a hard error.

**Pull request merged**